src/llm_responses.jl

```diff
@@ -0,0 +1,334 @@
+- `kwargs...`: Additional keyword arguments for the API call
+
+# Returns
+- `response`: The response from the OpenAI API
+"""
+function create_response(schema::AbstractResponseSchema, api_key::AbstractString,
+        model::AbstractString,
+        input;
+        previous_response_id::Union{Nothing, AbstractString} = nothing,
+        http_kwargs::NamedTuple = NamedTuple(),
+        streamcallback::Any = nothing,
+        tools::Vector{<:Any} = [],
+        api_kwargs::NamedTuple = NamedTuple(),
+        kwargs...)
+    # Build the request body
+    body = Dict{String, Any}(
+        "model" => model,
+        "input" => input
+    )
+
+    # Add optional parameters
+    !isnothing(previous_response_id) &&
+        (body["previous_response_id"] = previous_response_id)
+    !isempty(tools) && (body["tools"] = tools)
+
+    # Add streaming if callback provided
+    if !isnothing(streamcallback)
+        body["stream"] = true
+    end
+
+    # Add reasoning configuration from api_kwargs, default to detailed summary
+    # Valid summary values: "detailed", "concise", "none"
+    if haskey(api_kwargs, :reasoning)
+        body["reasoning"] = api_kwargs.reasoning
+    else
+        body["reasoning"] = Dict("summary" => "detailed")
+    end
+
+    # Add any other parameters from api_kwargs only
+    for (key, value) in api_kwargs
+        if key != :reasoning  # already handled above
+            body[string(key)] = value
+        end
+    end
+
+    # Make the API request
+    url = OpenAI.build_url(OpenAI.DEFAULT_PROVIDER, "responses")
+    headers = OpenAI.auth_header(OpenAI.DEFAULT_PROVIDER, api_key)
+
+    if !isnothing(streamcallback)
+        # Configure streaming callback - only pass schema, no extra kwargs
+        streamcallback, stream_kwargs = configure_callback!(streamcallback, schema)
+        
+        # Use streaming request
+        resp = streamed_request!(streamcallback, url, headers, body; http_kwargs...)
+        return OpenAI.OpenAIResponse(resp.status, JSON3.read(resp.body))
+    else
+        # Convert the body to JSON for non-streaming
+        json_body = JSON3.write(body)
+        
+        # Make the request
+        r = HTTP.request("POST", url, headers, json_body; http_kwargs...)
+        return OpenAI.OpenAIResponse(r.status, JSON3.read(r.body))
+    end
+end
+
+"""
+    render_responses(prompt; kwargs...) -> (input, instructions)
+
+Process different prompt types for the OpenAI Responses API and replace any placeholders
+with values from kwargs. The function has specialized methods for each supported prompt type.
+
+# Arguments
+- `prompt`: The prompt to process (String, Vector{<:AbstractMessage}, AITemplate, or Symbol)
+- `kwargs...`: Key-value pairs to replace placeholders in the prompt, e.g., {{key}} -> value
+
+# Returns
+- `input`: The input string for the API, with replacements
+- `instructions`: Optional instructions for the API, or nothing, with replacements
+"""
+function render_responses end
```

Comment:
can we stick to unified `render` function? It already has promptSchema as the first argument, so easy to extend with responsesSchema (see the signature elsewhere) ,eg, https://github.com/svilupp/PromptingTools.jl/blob/73d7039bea05df3b155740161ed2abb586f9a006/src/llm_shared.jl#L33